src/kontrol/foundry.py

```diff
@@ -503,25 +503,31 @@ def resolve_setup_proof_version(
         if reinit:
             if user_specified_setup_version is None:
                 _LOGGER.info(
-                    f'Creating a new version of test {test} because --reinit was specified and --setup-version is not specified.'
+                    f'Creating a new version of {test} because --reinit was specified and --setup-version is not specified.'
                 )
             elif not Proof.proof_data_exists(f'{test}:{user_specified_setup_version}', self.proofs_dir):
                 _LOGGER.info(
-                    f'Creating a new version of test {test} because --reinit was specified and --setup-version is set to a non-existing version'
+                    f'Creating a new version of {test} because --reinit was specified and --setup-version is set to a non-existing version'
+                )
+            elif not method.up_to_date(self.digest_file):
+                _LOGGER.info(
+                    f'Creating a new version of {test} because --reinit was specified and --setup-version is set to a dirty version'
```

Comment:
```suggestion
                    f'Creating a new version of {test} because --reinit was specified and --setup-version is set to an outdated version.'
```